_posts/2024-01-12-kasten-openshift-monitoring.markdown

```diff
@@ -0,0 +1,815 @@
+
+```
+cat<<EOF | oc create -f -
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  labels:
+    k8s-app: kasten-io
+  name: catalog
+  namespace: kasten-io
+spec:
+  endpoints:
+  - interval: 30s
+    port: http
+    scheme: http
+  selector:
+    matchLabels:
+      component: catalog
+EOF
+```
+
+After some time you should be able to query the catalog_actions_count metrics from the openshift monitoring service 
+
+```
+SECRET=`oc get secret -n openshift-user-workload-monitoring | grep  prometheus-user-workload-token | head -n 1 | awk '{print $1 }'`
+TOKEN=`echo $(oc get secret $SECRET -n openshift-user-workload-monitoring -o json | jq -r '.data.token') | base64 -d`
+THANOS_QUERIER_HOST=`oc get route thanos-querier -n openshift-monitoring -o json | jq -r '.spec.host'`
+curl -X GET -kG "https://$THANOS_QUERIER_HOST/api/v1/query?" --data-urlencode "query=catalog_actions_count" -H "Authorization: Bearer $TOKEN" |jq
+```
+
+You should get an answer like this one (Unless you did not create any kasten action): 
+
+```
+{
+  "status": "success",
+  "data": {
+    "resultType": "vector",
+    "result": [
+      {
+        "metric": {
+          "__name__": "catalog_actions_count",
+          "container": "catalog-svc",
+          "endpoint": "http",
+          "exported_namespace": "bnewell-pacman",
+          "instance": "10.131.1.77:8000",
+          "job": "catalog-svc",
+          "liveness": "live",
+          "namespace": "kasten-io",
+          "pod": "catalog-svc-56b97b4bfb-7vrzc",
+          "policy": "bnewell-pacman-backup",
+          "prometheus": "openshift-user-workload-monitoring/user-workload",
+          "service": "catalog-svc",
+          "status": "complete",
+          "type": "backup"
+        },
+        "value": [
+          1705579479.262,
+          "3"
+        ]
+      },
+      {
+        "metric": {
+          "__name__": "catalog_actions_count",
+          "container": "catalog-svc",
+          "endpoint": "http",
+          "exported_namespace": "bnewell-pacman",
+          "instance": "10.131.1.77:8000",
+          "job": "catalog-svc",
+          "liveness": "live",
+          "namespace": "kasten-io",
+          "pod": "catalog-svc-56b97b4bfb-7vrzc",
+          "policy": "bnewell-pacman-backup",
+
+          ..... Continue ....
+```
+
+
+The important part of the curl request is : 
+```
+"query=catalog_actions_count"
+```
+
+`catalog_action_count` is a metric exposed by the catalog, it give the status of the kasten actions. 
+
+# Detecting a failed action
+
+In order to test a failed action I'm going to fail a backup in one of the application 
+by creating a failed workload in the `mcourcy-pacman` namespace. Kasten will wait for the deployment to be ready and after a while it will fail the backup.
+
+`mcourcy-pacman` is a namespace which is protected by a policy. When I'll run this policy because of the failed workload the backup action will fail.
```

Comment:
```suggestion
`mcourcy-pacman` is a namespace which is protected by a [Kasten backup policy](https://docs.kasten.io/latest/api/policies.html). When I run this policy, the backup action will fail because of the failed workload.
```